### PR TITLE
Fix warning in ShmemTransport in some configurations

### PR DESCRIPTION
**Pull request by jwillemsen**
Extend CFT test with tests for set_expression_parameters

### CODE REVIEW

**Comment by jwillemsen** [comment]
These changes are complete

File: dds/DCPS/ContentFilteredTopicImpl.cpp [hunk 1]
```diff
@@ -30,7 +30,14 @@ ContentFilteredTopicImpl::ContentFilteredTopicImpl(const char* name,
   , filter_eval_(filter_expression, false /*allowOrderBy*/)
   , expression_parameters_(expression_parameters)
   , related_topic_(DDS::Topic::_duplicate(related_topic))
-{}
+{
+  if (DCPS_debug_level > 5) {
+      ACE_DEBUG((LM_DEBUG,
```
Comment (mitza-oci) [inline dds/DCPS/ContentFilteredTopicImpl.cpp]: Please check formatting in terms of level of indent and the placement of () for a function call.

File: dds/DCPS/ContentFilteredTopicImpl.cpp [hunk 2]
```diff
@@ -30,7 +30,14 @@ ContentFilteredTopicImpl::ContentFilteredTopicImpl(const char* name,
...
+                ACE_TEXT("(%P|%t) ContentFilteredTopicImpl::ContentFilteredTopicImpl() - ")
+                ACE_TEXT("Creating cft with filter <%C> which requires <%d> parameters\n"),
+                filter_expression, filter_eval_.number_parameters ()));
+  }
+}
```
Comment (mitza-oci) [inline dds/DCPS/ContentFilteredTopicImpl.cpp]: In this constructor the size of the expression_parameters is not checked using the same logic of set_expression_parameters.

File: dds/DCPS/ContentFilteredTopicImpl.cpp [hunk 3]
```diff
@@ -59,6 +66,21 @@ ContentFilteredTopicImpl::set_expression_parameters(const DDS::StringSeq& p)
     DDS::RETCODE_OUT_OF_RESOURCES);
 
   const CORBA::ULong len = p.length();
+
+  // Check sequence of strings that give values to the ‘parameters’ (i.e., "%n" tokens)
+  // in the filter_expression matches the size of the parameter sequence.
+  // The tokens start with 0 which means that when the maximum number used is 1 we need
+  // two parameters, (zero and one)
+  if (len != filter_eval_.number_parameters ()) {
```
Comment (mitza-oci) [inline dds/DCPS/ContentFilteredTopicImpl.cpp]: Should a sequence with more elements that required be allowed?  I don't feel strongly about it either way, and the spec doesn't help much.

File: tests/DCPS/ContentFilteredTopic/ContentFilteredTopicTest.cpp [hunk 4]
```diff
@@ -149,10 +149,26 @@ bool run_filtering_test(const DomainParticipant_var& dp,
...
+  DDS::StringSeq params_invalid(2);
+  params_invalid.length(2);
+  params_invalid[0] = "2";
+  params_invalid[1] = "2";
+  if (cft2->set_expression_parameters(params_invalid) != RETCODE_ERROR) {
+    cout << "ERROR: setting too much parameters should return an error"
+         << endl;
+    return false;
+  }
+
```
Comment (mitza-oci) [inline tests/DCPS/ContentFilteredTopic/ContentFilteredTopicTest.cpp]: would be good to also test too few parameters (more serious error), and also passing these sequences to DomainParticipant::create_contentfilteredtopic()

File: dds/DCPS/ContentFilteredTopicImpl.cpp (hunk 3 above)
Comment (jwillemsen) [inline dds/DCPS/ContentFilteredTopicImpl.cpp]: No, it should match, else you pass parameters that are not used

File: tests/DCPS/ContentFilteredTopic/ContentFilteredTopicTest.cpp (hunk 4 above)
Comment (jwillemsen) [inline tests/DCPS/ContentFilteredTopic/ContentFilteredTopicTest.cpp]: Added more test cases

Comment (jwillemsen) [inline tests/DCPS/ContentFilteredTopic/ContentFilteredTopicTest.cpp]: done

**Comment by jwillemsen** [comment]
@mitza-oci These changes are ready, still working on trying to reproduce the runtime crashes we see with AXCIOMA and OpenDDS

File: dds/DCPS/transport/shmem/ShmemTransport.cpp [hunk 5]
```diff
@@ -116,6 +115,7 @@ ShmemTransport::configure_i(TransportInst* config)
                     ACE_TEXT("no platform support for shared memory!\n")),
                    false);
 #else
+  ShmemInst* conf = dynamic_cast<ShmemInst*>(config);
```
Comment (mitza-oci) [inline dds/DCPS/transport/shmem/ShmemTransport.cpp]: This appears to be unrelated to the PR.

Comment (jwillemsen) [inline dds/DCPS/transport/shmem/ShmemTransport.cpp]: It fixes a compile warning I did get in one of our test builds when testing this change, updated the PR title to reflect that

Comment (mitza-oci) [inline dds/DCPS/transport/shmem/ShmemTransport.cpp]: That's fine to change the title, and I don't think it's worth the effort to go through and separate this now, but please keep in mind for the future that it's much cleaner to handle these submissions as distinct PRs with a single purpose each.

Comment (jwillemsen) [inline dds/DCPS/transport/shmem/ShmemTransport.cpp]: Will try to separate things as much as possible in distinct PRs in the future
